components/keys/src/types.rs

```diff
@@ -35,24 +39,26 @@ impl Key {
     #[inline]
     pub fn from_raw(key: &[u8]) -> Key {
         // adding extra length for appending timestamp
-        let len = codec::bytes::max_encoded_bytes_size(key.len()) + codec::number::U64_SIZE;
+        let len = MemComparableByteCodec::encoded_len(key.len()) + number::U64_SIZE;
         let mut encoded = Vec::with_capacity(len);
-        encoded.encode_bytes(key, false).unwrap();
+        encoded.write_comparable_bytes(key).unwrap();
         Key(encoded)
     }
 
     /// Gets and moves the raw representation of this key.
     #[inline]
     pub fn into_raw(self) -> Result<Vec<u8>, codec::Error> {
         let mut k = self.0;
-        bytes::decode_bytes_in_place(&mut k, false)?;
+        let (_read_bytes, written_bytes) =
+            MemComparableByteCodec::try_decode_first_in_place(&mut k)?;
+        k.truncate(written_bytes);
         Ok(k)
     }
 
     /// Gets the raw representation of this key.
     #[inline]
     pub fn to_raw(&self) -> Result<Vec<u8>, codec::Error> {
-        bytes::decode_bytes(&mut self.0.as_slice(), false)
+        self.0.as_slice().read_comparable_bytes()
```

Comment:
maybe `self.0.read_comparable_bytes()` is enough?

Reply:
we can't
because error[E0599]: no method named `read_comparable_bytes` found for type `std::vec::Vec<u8>` in the current scope